### PR TITLE
[Integration][ADO] Bug Fix: Default mapping errors

### PR DESCRIPTION
**Pull request by oiadebayo**
# Description

**What -**
This PR addresses a few bugs and improvements in the Azure DevOps (ADO) integration related to the default mappings for service and work-item entities, handling of project data, and logging enhancements.

**Why -**
- The original default mapping for services caused errors when repository names contained special characters like parentheses. The error prevented the repositories from being imported into Port.
- The work-item mapping had an issue where the relation to the project was incorrectly based on `System.TeamProject`, which referenced the project name instead of the internal project identifier
- It was necessary to update the mapping so that full project data, not just the project ID, is passed to the work-item entities
- Additionally, to aid in debugging and monitoring, general logs should be displayed when the integration is querying the work items.

 **How -**
- The mapping for services was updated to removing additional special characters that could cause the repository name to fail. 
- The work-item mapping was updated to reference the `__projectId`, extracting the correct internal project identifier, ensuring accurate associations in Port.
- The integration was improved to pass full project data to work-item entities instead of just the project ID
- Added general logging to the integration to display information when querying work items, which will help in tracking the integration's activity and diagnosing issues.

## Type of change

Please leave one option from the following and delete the rest:

- [x] Bug fix (non-breaking change which fixes an issue)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync finishes successfully
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [x] Integration able to create all default resources from scratch
- [x] Resync able to create entities
- [x] Resync able to update entities
- [x] Resync able to detect and delete entities
- [x] Resync finishes successfully
- [ ] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [ ] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [ ] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: integrations/azure-devops/.port/resources/port-app-config.yaml [hunk 1]
```diff
@@ -1,77 +1,78 @@
-createMissingRelatedEntities: true
 deleteDependentEntities: true
+createMissingRelatedEntities: true
 resources:
   - kind: project
     selector:
       query: 'true'
-      defaultTeam: "true"
+      defaultTeam: true
```
Comment (PeyGis) [inline integrations/azure-devops/.port/resources/port-app-config.yaml:7]: shouldn't this be 'true'?

File: integrations/azure-devops/azure_devops/client/azure_devops_client.py [hunk 2]
```diff
@@ -218,11 +221,12 @@ async def _fetch_work_items_in_batches(
...
             work_items_response.raise_for_status()
             work_items.extend(work_items_response.json()["value"])
+
         return work_items
 
     def _add_project_id_to_work_items(
-        self, work_items: list[dict[str, Any]], project_id: str
-    ) -> None:
+        self, work_items: list[dict[str, Any]], project_id: str, project: dict[str, Any]
+    ) -> list[dict[str, Any]]:
```
Comment (PeyGis) [inline integrations/azure-devops/azure_devops/client/azure_devops_client.py:229]: rename the function to __add_project_details_to_work_items(self, work_item, project)

then you can do 
__projectId = project["id]
 __project = project

**Review by PeyGis — CHANGES_REQUESTED** [review]
left some comments

File: integrations/azure-devops/azure_devops/client/azure_devops_client.py (hunk 2 above)
Comment (oiadebayo) [reply integrations/azure-devops/azure_devops/client/azure_devops_client.py]: Done

File: integrations/azure-devops/.port/resources/port-app-config.yaml (hunk 1 above)
Comment (oiadebayo) [reply integrations/azure-devops/.port/resources/port-app-config.yaml]: Done

File: integrations/azure-devops/azure_devops/client/azure_devops_client.py [hunk 3]
```diff
@@ -155,15 +154,18 @@ async def generate_work_items(self) -> AsyncGenerator[list[dict[str, Any]], None
...
-
+                logger.info(
+                    f"Found {len(work_item_ids)} work item IDs for project {project['name']}"
+                )
                 # 2. Fetch work items using the IDs (in batches if needed)
                 work_items = await self._fetch_work_items_in_batches(
                     project["id"], work_item_ids
                 )
-
+                logger.info(f"Received {len(work_items)} work items")
```
Comment (PeyGis) [inline integrations/azure-devops/azure_devops/client/azure_devops_client.py:164]:
```suggestion
                logger.debug(f"Received {len(work_items)} work items")

```

**Review by PeyGis — APPROVED** [review]
LGTM

**Review by Tankilevitch — APPROVED** [review]

File: integrations/azure-devops/.port/resources/port-app-config.yaml [hunk 4]
```diff
@@ -1,77 +1,78 @@
...
+            description: .fields."System.Description"
+            link: .url
+            reason: .fields."System.Reason"
+            createdBy: .fields."System.CreatedBy".displayName
+            changedBy: .fields."System.ChangedBy".displayName
+            createdDate: .fields."System.CreatedDate"
+            changedDate: .fields."System.ChangedDate"
           relations:
-            project: '.fields."System.TeamProject" | gsub(" "; "")'
+            project: .__projectId
```
Comment (Tankilevitch) [inline integrations/azure-devops/.port/resources/port-app-config.yaml:78]: This should be also updated to be without special characters?

Comment (PeyGis) [reply integrations/azure-devops/.port/resources/port-app-config.yaml:78]: good catch

Comment (PeyGis) [inline integrations/azure-devops/.port/resources/port-app-config.yaml:78]:
```suggestion
            project: .__projectId | gsub(" "; "")
```

**Review by PeyGis — APPROVED** [review]

**Pull request merged**